s4e-backend/src/main/java/pl/cyfronet/s4e/data/repository/SceneRepositoryImpl.java

```diff
@@ -0,0 +1,352 @@
+        queryByCloudCover(params, resultQuery);
+    }
+
+    private void timeParameters(PreparedStatement ps, Map<String, Object> params) throws SQLException {
+        parametersBySensingTime(params, ps);
+        parametersByIngestionTime(params, ps);
+    }
+
+    private void textParameters(PreparedStatement ps, Map<String, Object> params) throws SQLException {
+        parametersBySatellitePlatform(params, ps);
+        parametersByProductType(params, ps);
+        parametersByPolarisation(params, ps);
+        parametersByProcessingLevel(params, ps);
+
+        parametersBySensorMode(params, ps);
+        parametersByCollection(params, ps);
+        parametersByTimeliness(params, ps);
+        parametersByInstrument(params, ps);
+        parametersByProductLevel(params, ps);
+        parametersByRelativeOrbitNumber(params, ps);
+        parametersByAbsoluteOrbitNumber(params, ps);
+    }
+
+    private void numberParameters(PreparedStatement ps, Map<String, Object> params) throws SQLException {
+        //float
+        parametersByCloudCover(params, ps);
+    }
+
+    private void queryBySensingTime(Map<String, Object> params, StringBuilder resultQuery) {
+        if (params.containsKey("sensingFrom") && params.containsKey("sensingTo")) {
+            // [TIMESTAMP] sensingFrom / sensingTo -> sensing_time
+            resultQuery.append(" AND to_timestamp(metadata_content->>'sensing_time', 'YYYY-MM-DD\"T\"HH24:MI:SS\"Z\"')");
+            resultQuery.append(" BETWEEN ? AND ? ");
+        } else {
+            if (params.containsKey("sensingFrom")) {
+                resultQuery.append(" AND to_timestamp(metadata_content->>'sensing_time', 'YYYY-MM-DD\"T\"HH24:MI:SS\"Z\"') >= ? ");
+            }
+            if (params.containsKey("sensingTo")) {
+                resultQuery.append(" AND to_timestamp(metadata_content->>'sensing_time', 'YYYY-MM-DD\"T\"HH24:MI:SS\"Z\"') <= ? ");
+            }
+        }
+    }
+
+    private void parametersBySensingTime(Map<String, Object> params, PreparedStatement ps) throws SQLException {
+        if (params.containsKey("sensingFrom") && params.containsKey("sensingTo")) {
+            ps.setObject(paramIndex++, params.get("sensingFrom"));
+            ps.setObject(paramIndex++, params.get("sensingTo"));
+        } else {
+            if (params.containsKey("sensingFrom")) {
+                ps.setObject(paramIndex++, params.get("sensingFrom"));
+            }
+            if (params.containsKey("sensingTo")) {
+                ps.setObject(paramIndex++, params.get("sensingTo"));
+            }
+        }
+    }
+
+    private void queryByIngestionTime(Map<String, Object> params, StringBuilder resultQuery) {
+        if (params.containsKey("ingestionFrom") && params.containsKey("ingestionTo")) {
+            // [TIMESTAMP] ingestionFrom / ingestionTo -> ingestion_time
+            resultQuery.append(" AND to_timestamp(metadata_content->>'ingestion_time', 'YYYY-MM-DD\"T\"HH24:MI:SS\"Z\"')");
```

Comment:
`to_timestamp(metadata_content->>'ingestion_time', 'YYYY-MM-DD\"T\"HH24:MI:SS\"Z\"')` can be changed to function

return to_timestamp(metadata_content->>${field}, 'YYYY-MM-DD\"T\"HH24:MI:SS\"Z\"')